src/integrity-checker.js

```diff
@@ -118,23 +109,57 @@ export default class InstallationIntegrityChecker {
   }
 
   /**
-   * returns a list of files recursively in a directory sorted
+   * Get the list of the directories that contain our modules (there might be multiple such folders b/c of workspaces).
+   */
+
+  _getModulesFolders({workspaceLayout}: {workspaceLayout: ?WorkspaceLayout} = {}): Array<string> {
+    const locations = [];
+
+    if (this.config.modulesFolder) {
+      locations.push(this.config.modulesFolder);
+    } else {
+      locations.push(path.join(this.config.lockfileFolder, constants.NODE_MODULES_FOLDER));
+    }
+
+    if (workspaceLayout) {
+      for (const workspaceName of Object.keys(workspaceLayout.workspaces)) {
+        const loc = workspaceLayout.workspaces[workspaceName].loc;
+
+        if (loc) {
```

Comment:
I think after my fixes, we don't need the emptiness check here anymore.

Referring to these changes:
 - https://github.com/yarnpkg/yarn/pull/4129/files#diff-ffb8cf10e4b2d16cc7df603574698af6R287
 - https://github.com/yarnpkg/yarn/pull/4131/files#diff-1ff3de58ebf278d554d14bcbe160d981R12